From 929ee78e8bda4e63ad42d764e8eca1b2a36c3383 Mon Sep 17 00:00:00 2001
From: Evan Maddock <maddock.evan@vivaldi.net>
Date: Wed, 2 Aug 2023 13:09:09 -0400
Subject: [PATCH] keyboard-layout: Don't try to initialize IBus

IBus already gets initialized in wm, and calling that function here causes severe breakage, leading to budgie-panel being repeatedly killed for OOM.

Signed-off-by: Evan Maddock <maddock.evan@vivaldi.net>
---
 .../applets/keyboard-layout/KeyboardLayoutApplet.vala    | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala b/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala
index 3f04d947..325331e7 100644
--- a/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala
+++ b/src/panel/applets/keyboard-layout/KeyboardLayoutApplet.vala
@@ -30,7 +30,6 @@ errordomain InputMethodError {
 class AppletIBusManager : GLib.Object {
 	private HashTable<string,weak IBus.EngineDesc> engines = null;
 	private List<IBus.EngineDesc>? enginelist = null;
-	private bool did_ibus_init = false;
 	private bool ibus_available = true;
 	private IBus.Bus? bus = null;
 
@@ -94,14 +93,6 @@ class AppletIBusManager : GLib.Object {
 	 * We gained connection to the ibus daemon
 	 */
 	private void ibus_connected() {
-		/*
-		 * Init ibus if necessary, to ensure we have the types available to
-		 * glib. After this, try and gain all the engines
-		 */
-		if (!did_ibus_init) {
-			IBus.init();
-			did_ibus_init = true;
-		}
 		this.bus.list_engines_async.begin(-1, null, on_engines_get);
 	}
 
